test/werewolf/engine/state_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+
+module Werewolf
+  module Engine
+    class StateTest < Minitest::Test
+      def test_starts_at_correct_state
+        assert_equal State.new.current, State::STATES.first
+      end
+
+      def test_starts_in_transition_state
+        assert State.new.transitioning?
+      end
+
+
+      def test_starts_at_correct_turn
+        assert_equal State.new.turn_number, 0
+      end
+
+
+      def test_starts_at_correct_day
+        assert_equal State.new.day_number, 0
+      end
+
+      def test_next_advances_the_state
+        state = State.new
+
+        assert_equal state.current, :dawn
+        assert_equal state.next, :day
+        assert_equal state.current, :day
+        assert_equal state.next, :dusk
+        assert_equal state.next, :night
+        assert_equal state.next, :dawn
+        assert_equal state.next, :day
+      end
+
+
+      def test_next_advances_the_counters
+        state = State.new
+
+        state.enumerator.expects(:next).times(8)
```

Comment:
i don't think you need this.  ideally we're looking to test _behavior_, not implementation.  this is purely testing implementation, and you could just as easily test behavior.  You call `next` 8 times and the day and turn should be right, perfect.  Tying the test to the actual internals just makes the test more fragile when you refactor